app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/SelectBasisForClaimController.scala

```diff
@@ -56,102 +57,91 @@ class SelectBasisForClaimController @Inject() (
+      withAnswersAndRoutes[BasisOfClaim] { (fillingOutClaim, answer, router) =>
+        val emptyForm  = SelectBasisForClaimController.reasonForClaimForm
+        val filledForm = answer.fold(emptyForm)(basisOfClaim => emptyForm.fill(SelectReasonForClaim(basisOfClaim)))
+        Ok(
+          selectReasonForClaimPage(
+            filledForm,
+            getPossibleClaimTypes(fillingOutClaim.draftClaim, journey),
+            getBasisOfClaimsExamples(fillingOutClaim.draftClaim, journey),
+            isAmend,
+            router
+          )
+        )
       }
     }
 
-  def selectBasisForClaimSubmit(journey: JourneyBindable): Action[AnyContent] = submit(false)(journey)
-  def changeBasisForClaimSubmit(journey: JourneyBindable): Action[AnyContent] = submit(true)(journey)
+  def selectBasisForClaimSubmit(journey: JourneyBindable): Action[AnyContent] = submit(isAmend = false)(journey)
+  def changeBasisForClaimSubmit(journey: JourneyBindable): Action[AnyContent] = submit(isAmend = true)(journey)
 
   def submit(isAmend: Boolean)(implicit journey: JourneyBindable): Action[AnyContent] =
     authenticatedActionWithSessionData.async { implicit request =>
-      withAnswersAndRoutes[BasisOfClaimAnswer] { (fillingOutClaim, _, router) =>
-        val isMrnJourney = fillingOutClaim.draftClaim.isMrnFlow
+      withAnswersAndRoutes[BasisOfClaim] { (fillingOutClaim, _, router) =>
         SelectBasisForClaimController.reasonForClaimForm
           .bindFromRequest()
           .fold(
             formWithErrors =>
-              BadRequest(selectReasonForClaimPage(formWithErrors, allClaimsTypes, isAmend, isMrnJourney, router)),
+              BadRequest(
+                selectReasonForClaimPage(
+                  formWithErrors,
+                  getPossibleClaimTypes(fillingOutClaim.draftClaim, journey),
+                  getBasisOfClaimsExamples(fillingOutClaim.draftClaim, journey),
+                  isAmend,
+                  router
+                )
+              ),
             formOk => {
-              val updatedBasisClaim = CompleteBasisOfClaimAnswer(formOk.reasonForClaim)
-              val newDraftClaim     =
-                fillingOutClaim.draftClaim
-                  .fold(_.copy(basisOfClaimAnswer = Option(updatedBasisClaim), reasonForBasisAndClaimAnswer = None))
-              val updatedJourney    = fillingOutClaim.copy(draftClaim = newDraftClaim)
+              val updatedJourney = FillingOutClaim.of(fillingOutClaim)(
+                _.copy(
+                  basisOfClaimAnswer = formOk.reasonForClaim.some,
+                  reasonForBasisAndClaimAnswer = None
+                )
+              )
 
               EitherT
-                .liftF(updateSession(sessionStore, request)(_.copy(journeyStatus = Some(updatedJourney))))
+                .liftF(updateSession(sessionStore, request)(_.copy(journeyStatus = updatedJourney.some)))
```

Comment:
liftF is unnecessary. updateSession returns Future[Either].
So you can just do EitherT(updateSession), and fix the leftMap ...

Reply:
Good to know. This's the old code which I haven't touched. As far as I know there a lots places like this, perhaps it makes sense to refactored it at once extracting it to some common service.

Reply:
You can get into type inference issues by just using the `apply` method on `EItherT` hence the use of `liftF` - both have the same semantics.